make.js

```diff
@@ -413,6 +416,26 @@ target.publish = function() {
     var server = options.server;
     assert(server, 'server');
 
+    // get the branch/commit info
+    var refs = util.getRefs();
+
+    // test whether to publish the non-aggregated tasks zip
+    var release = refs.head.release;
+    var commit = refs.heads.commit;
+    if (!release ||
+        !refs.releases[release] ||
+        commit != refs.releases[release].commit) {
+
+        console.log(`##vso[task.logissue type=warning]Skipping publish for non-aggregated tasks zip. HEAD is not the tip of a release branch.`);
+    }
+    else {
+        var nonAggregatedZipPath = path.join(packagePath, 'non-aggregated-tasks.zip');
+        util.storeNonAggregatedZip(nonAggregatedZipPath, release, commit);
+    }
+    // todo: store the non-aggregated tasks zip
+
+// This step will only be performed if the current branch is the tip of a release branch (compare against origin branches). Otherwise if the current branch is master, then no-op. And if anything else, error – the “publish” build target should never be run for a user or feature branch.
```

Comment:
fix logic further below when creating the nupkg to pull in the aggregated layout